scipy/stats/_multicomp.py

```diff
@@ -0,0 +1,311 @@
+
+    >>> import numpy as np
+    >>> control = np.array([7.40, 8.50, 7.20, 8.24, 9.84, 8.32])
+    >>> drug_a = np.array([9.76, 8.80, 7.68, 9.36])
+    >>> drug_b = np.array([12.80, 9.68, 12.16, 9.20, 10.55])
+
+    The `dunnett` statistic is sensitive to the difference in means between
+    the samples.
+
+    We would like to see if the means between any of the groups are
+    significantly different. First, visually examine a box and whisker plot.
+
+    >>> import matplotlib.pyplot as plt
+    >>> fig, ax = plt.subplots(1, 1)
+    >>> ax.boxplot([control, drug_a, drug_b])
+    >>> ax.set_xticklabels(["Control", "Drug A", "Drug B"])  # doctest: +SKIP
+    >>> ax.set_ylabel("mean")  # doctest: +SKIP
+    >>> plt.show()
+
+    From the box and whisker plot, we can see overlap in the interquartile
+    ranges between the control group and the group from drug A.
+    We can apply the `dunnett`
+    test to determine if the difference between means is significant. We
+    set a significance level of .05 to reject the null hypothesis.
+
+    >>> from scipy.stats import dunnett
+    >>> res = dunnett(drug_a, drug_b, control=control)
+    >>> res.pvalue
+    array([0.47773146, 0.00889328])  # random
+
+    The null hypothesis is that each group has the same mean. The p-value for
+    comparisons between ``control`` and ``drug_b`` do not exceed .05,
+    so we reject the null hypothesis that they
+    have the same means. The p-value of the comparison between ``control``
+    and ``drug_a`` exceeds .05, so we accept the null hypothesis that there
+    is not a significant difference between their means.
+
+    """
+    rng = check_random_state(random_state)
+    control = np.asarray(control)
+
+    rho, df, n_group = iv_dunnett(
+        observations=observations, control=control
+    )
+
+    with suppress_warnings() as sup:
+        sup.filter(RuntimeWarning)
+        statistic = np.array([
+            stats.ttest_ind(
+                obs_, control, alternative=alternative, random_state=rng
+            ).statistic
+            for obs_ in observations
+        ])
+
+    pvalue = pvalue_dunnett(
+        rho=rho, df=df,
+        statistic=statistic, alternative=alternative,
+        rng=rng
+    )
+
+    control_mean = np.mean(control)
+    observations_mean = np.array([np.mean(obs_) for obs_ in observations])
+    std = np.sqrt((
+        np.sum(control**2)
+        + np.sum([obs_**2 for group in observations for obs_ in group])
+        - n_group*(control_mean**2 + np.sum(observations_mean**2))
+    ) / df)
+
+    return DunnettResult(
+        statistic=statistic, pvalue=pvalue,
+        _rho=rho, _df=df, _std=std,
+        _observations_mean=observations_mean,
+        _control_mean=control_mean
+    )
+
+
+def iv_dunnett(
+    observations: npt.ArrayLike, control: npt.ArrayLike
+) -> tuple[np.ndarray, int, int]:
+    """Specific parameters for Dunnett's test.
+
+    Covariance matrix depends on the number of observations in each group:
```

Comment:
Looks like there is just one formula, so I think that mention of these special cases can be deleted.

Reply:
Ok I added that so we have a ref about what balanced and unbalanced designs are.